modules/openid-federation-common/src/jvmMain/kotlin/com/sphereon/oid/fed/common/jwt/JoseJwt.jvm.kt

```diff
@@ -0,0 +1,61 @@
+package com.sphereon.oid.fed.common.jwt
+
+import com.nimbusds.jose.JWSAlgorithm
+import com.nimbusds.jose.JWSHeader
+import com.nimbusds.jose.JWSSigner
+import com.nimbusds.jose.JWSVerifier
+import com.nimbusds.jose.crypto.RSASSASigner
+import com.nimbusds.jose.crypto.RSASSAVerifier
+import com.nimbusds.jose.jwk.RSAKey
+import com.nimbusds.jose.jwk.gen.RSAKeyGenerator
+import com.nimbusds.jwt.JWTClaimsSet
+import com.nimbusds.jwt.SignedJWT
+import java.util.*
+
+actual fun sign(
+    payload: String,
+    opts: MutableMap<String, Any>?
+): String {
+    var rsaJWK = opts?.get("key") as RSAKey?
+    val kid = rsaJWK?.keyID ?: UUID.randomUUID().toString()
+    val header: JWSHeader?
+    if (opts?.get("jwtHeader") != null) {
+        header = JWSHeader.parse(opts["jwtHeader"] as String?)
+    } else {
+        header = JWSHeader.Builder(JWSAlgorithm.RS256).keyID(kid).build()
+    }
+
+    if (rsaJWK == null) {
+        rsaJWK = RSAKeyGenerator(2048)
+            .keyID(kid)
+            .generate()
+    }
```

Comment:
from row 19
````
val rsaJWK = opts?.get("key") as? RSAKey ?: RSAKeyGenerator(2048)
    .keyID(UUID.randomUUID().toString())
    .generate()

val header = opts?.get("jwtHeader")?.let { JWSHeader.parse(it as String) }
    ?: JWSHeader.Builder(JWSAlgorithm.RS256).keyID(rsaJWK.keyID).build()
````
    
?
(But formatted a bit better)

Reply:
Nice one, now it looks more like Kotlin. Changed applied